tinymce-angular-component/src/main/ts/utils/ScriptLoader.ts

```diff
@@ -15,41 +15,61 @@ export interface IStateObj {
   scriptLoaded: boolean;
 }
 
-const injectScriptTag = (scriptId: string, doc: Document, url: string, callback: callbackFn) => {
-  const scriptTag = doc.createElement('script');
-  scriptTag.referrerPolicy = 'origin';
-  scriptTag.type = 'application/javascript';
-  scriptTag.id = scriptId;
-  scriptTag.addEventListener('load', callback);
-  scriptTag.src = url;
-  if (doc.head) {
-    doc.head.appendChild(scriptTag);
-  }
-};
-
-const create = (): IStateObj => {
+const createState = (): IStateObj => {
   return {
     listeners: [],
     scriptId: uuid('tiny-script'),
     scriptLoaded: false
   };
 };
 
-const load = (state: IStateObj, doc: Document, url: string, callback: callbackFn) => {
-  if (state.scriptLoaded) {
-    callback();
-  } else {
-    state.listeners.push(callback);
-    if (!doc.getElementById(state.scriptId)) {
-      injectScriptTag(state.scriptId, doc, url, () => {
-        state.listeners.forEach((fn) => fn());
-        state.scriptLoaded = true;
-      });
+interface ScriptLoader {
+  load: (doc: Document, url: string, callback: callbackFn) => void;
+  reinitialize: () => void;
+}
+
+const CreateScriptLoader = (): ScriptLoader => {
+  let state: IStateObj = createState();
+
+  const injectScriptTag = (scriptId: string, doc: Document, url: string, callback: callbackFn) => {
+    const scriptTag = doc.createElement('script');
+    scriptTag.referrerPolicy = 'origin';
+    scriptTag.type = 'application/javascript';
+    scriptTag.id = scriptId;
+    scriptTag.addEventListener('load', callback);
```

Comment:
This will potentially cause a memory leak since we don't unbind the load event on the script tag when the callback completes.

- [ ] Make sure to unbind the `load` event once complete